controllers/ingressnodefirewall_controller.go

```diff
@@ -362,10 +367,7 @@ func (r *IngressNodeFirewallReconciler) buildNodeStates(
 }
 
 func isLegalInterfaceName(iface string) bool {
-	if iface == "" {
-		return false
-	}
-	return true
+	return interfaces.IsValidInterfaceNameAndState(iface)
```

Comment:
isLegalInterfaceName runs as part of the IngressNodeFirewallReconsiler which is part of the operator pod. But the operator can't do a local lookup for valid interfaces names. 
a) the operator runs without host network
b) even if it did, the interfaces on the operator's host might be different from those on the other hosts. 

So I think here we need a very generic verificatoin for valid syntax for interface names. And then inside the ingressnodefirewallnodestate controller that runs inside the daemon we need a check for IsValidInterfaceNameAndState. If that there returns false, we must post a status to the NodeState resource or emit an event or something else.

Reply:
I moved the check to the daemon pkg and return an error when condition is detected

Reply:
I moved that check to the webhook and added more check to make sure is legit interface(s) name